cunumeric/_ufunc/ufunc.py

```diff
@@ -349,6 +527,17 @@ def _resolve_dtype(self, arrs, orig_args, casting, precision_fixed):
                     chosen = in_dtypes
                     break
 
+            # If there's no safe match and the operands have different types,
```

Comment:
Is this block specifically for the case where the second argument is of `np.int64` type? If so, could we instead get away with just adding cases for `l` to the `types` on `ldexp`? NumPy seems to advertise support for combinations like `(np.float32, np.int64) -> np.float32`:

```
>>> np.ldexp.types
['ei->e', 'fi->f', 'el->e', 'fl->f', 'di->d', 'dl->d', 'gi->g', 'gl->g']
```

Reply:
> Is this block specifically for the case where the second argument is of np.int64 type?

No, that's for cases where the type resolution fails in the first round of lookup. `ldexp` is the only operation whose operands have different types so it's going to be the only beneficiary of this change at the moment.

> If so, could we instead get away with just adding cases for l to the types on ldexp? 

It's not that simple. Such type overloading is a bit problematic because of the way we do type dispatch on the C++ side. To support the overloading, we'd need to do a double dispatch on the operand types or at least special case those operators, both of which seems too big a change just to support this single operator with dubious merit. I suggest we stick to what we have and revisit this later if we find actual users who want it.

Also note that those "extra" ops are in some sense fused ops., as the underlying math library function (`std::ldexp`) any of these NumPy implementations would use only takes a 32-bit integer; so those extra operations are essentially an `std::ldexp` call fused with type conversion. Having such fused ops is itself not a problem, but I think we need a more disciplined approach than arbitrarily choosing to fuse only some of them.